### PR TITLE
test(integration): add tests with TLS registry

### PR DESCRIPTION
**Pull request by knqyf263**
Ref. https://github.com/aquasecurity/harbor-scanner-trivy/pull/85

### CODE REVIEW

File: integration/registry_test.go [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+	"os/exec"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/docker/docker/client"
+	"github.com/docker/go-connections/nat"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	testcontainers "github.com/testcontainers/testcontainers-go"
```
Comment (danielpacak) [inline integration/registry_test.go:20]: NIT: This shows as redundant alias in my IDE

File: integration/registry_test.go [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+	"github.com/testcontainers/testcontainers-go/wait"
+
+	"github.com/aquasecurity/fanal/analyzer"
+	"github.com/aquasecurity/fanal/cache"
+	"github.com/aquasecurity/fanal/extractor/docker"
+	testdocker "github.com/aquasecurity/fanal/integration/docker"
+	"github.com/aquasecurity/fanal/types"
+
+	_ "github.com/aquasecurity/fanal/analyzer/command/apk"
```
Comment (danielpacak) [inline integration/registry_test.go:29]: NIT: There're 2 places where you have to import all analyzers. Maybe consider aggregating them:

```go
// fanal/analyzer/library/analyzer/analyzers.go

package analyzer

import (
	// Initialize all knows analyzers
	_ "github.com/aquasecurity/fanal/analyzer/command/apk"
	_ "github.com/aquasecurity/fanal/analyzer/library/bundler"
	_ "github.com/aquasecurity/fanal/analyzer/library/cargo"
	_ "github.com/aquasecurity/fanal/analyzer/library/composer"
	_ "github.com/aquasecurity/fanal/analyzer/library/npm"
	_ "github.com/aquasecurity/fanal/analyzer/library/pipenv"
	_ "github.com/aquasecurity/fanal/analyzer/library/poetry"
	_ "github.com/aquasecurity/fanal/analyzer/library/yarn"
	_ "github.com/aquasecurity/fanal/analyzer/os/alpine"
	_ "github.com/aquasecurity/fanal/analyzer/os/amazonlinux"
	_ "github.com/aquasecurity/fanal/analyzer/os/debianbase"
	_ "github.com/aquasecurity/fanal/analyzer/os/photon"
	_ "github.com/aquasecurity/fanal/analyzer/os/redhatbase"
	_ "github.com/aquasecurity/fanal/analyzer/os/suse"
	_ "github.com/aquasecurity/fanal/analyzer/pkg/apk"
	_ "github.com/aquasecurity/fanal/analyzer/pkg/dpkg"
	_ "github.com/aquasecurity/fanal/analyzer/pkg/rpmcmd"
)
```

Then you just need a single import here and in the main.go:

```go
import (
	_ "github.com/aquasecurity/fanal/analyzer"
)
```

File: integration/registry_test.go [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+		{
+			name:      "sad path: tls verify",
+			imageName: "alpine:3.10",
+			imageFile: "testdata/fixtures/alpine-310.tar.gz",
+			option: types.DockerOption{
+				Timeout:  60 * time.Second,
+				UserName: registryUsername,
+				Password: registryPassword,
+			},
+			expectedOS: types.OS{Name: "3.10.2", Family: "alpine"},
```
Comment (danielpacak) [inline integration/registry_test.go:141]: No need to specify OS expectation as you expect err?

File: integration/registry_test.go [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+		},
+		{
+			name:      "sad path: no credential",
+			imageName: "alpine:3.10",
+			imageFile: "testdata/fixtures/alpine-310.tar.gz",
+			option: types.DockerOption{
+				Timeout:               60 * time.Second,
+				InsecureSkipTLSVerify: true,
+			},
+			expectedOS: types.OS{Name: "3.10.2", Family: "alpine"},
```
Comment (danielpacak) [inline integration/registry_test.go:152]: No need to specify OS expectation as you expect err?

File: integration/registry_test.go (hunk 1 above)
Comment (knqyf263) [reply integration/registry_test.go:20]: Yes, I know. But my IDE repeatedly inserts and deletes the import if I don't add `testcontainers`. It's weird, but I have no idea to fix this issue...
goland

File: integration/registry_test.go (hunk 2 above)
Comment (knqyf263) [reply integration/registry_test.go]: Good catch! I fixed it. https://github.com/aquasecurity/fanal/pull/99/commits/4ad565f37a916158143a88ea270cba21eb2168dd

File: integration/registry_test.go (hunk 3 above)
Comment (knqyf263) [reply integration/registry_test.go]: Thanks. I removed it.
https://github.com/aquasecurity/fanal/pull/99/commits/dc13b099a7934c42daba188eebe84b1bf8c01851

File: integration/registry_test.go (hunk 4 above)
Comment (knqyf263) [reply integration/registry_test.go]: Ditto
https://github.com/aquasecurity/fanal/pull/99/commits/dc13b099a7934c42daba188eebe84b1bf8c01851

File: integration/registry_test.go (hunk 1 above)
Comment (simar7) [reply integration/registry_test.go:20]: FWIW @danielpacak  and @knqyf263, I have the same behaviour as seen here. I tried very hard to get rid of it too but it won't. What IDE are you running @danielpacak? What version?

Comment (danielpacak) [reply integration/registry_test.go:20]: > FWIW @danielpacak and @knqyf263, I have the same behaviour as seen here. I tried very hard to get rid of it too but it won't. What IDE are you running @danielpacak? What version?

I'm using Intellij IDEA ULTIMATE 2019.3.4 with JetBrains Go plugin v193.6494.35.125

Comment (simar7) [reply integration/registry_test.go:20]: I guess the behaviour is different across IDEs. I'm using Goland v2019.3.4

**Pull request merged**